frontend/app/src/components/asset-manager/MergeDialog.vue

```diff
@@ -5,46 +5,47 @@ import { toMessages } from '@/utils/validation';
 
 defineProps<{ value: boolean }>();
 
-const emit = defineEmits(['input']);
+const emit = defineEmits<{ (e: 'input', value: boolean): void }>();
 const done = ref(false);
-const errorMessages = ref<string[]>([]);
-const target = ref('');
-const source = ref('');
+const errorMessages = ref<
+  Partial<Record<'targetIdentifier' | 'sourceIdentifier', string[]>>
+>({});
+const targetIdentifier = ref('');
+const sourceIdentifier = ref('');
 const pending = ref(false);
 
 const { mergeAssets } = useAssets();
 const { t } = useI18n();
 
 const reset = () => {
   set(done, false);
-  set(target, '');
-  set(source, '');
+  set(targetIdentifier, '');
+  set(sourceIdentifier, '');
   set(pending, false);
-  set(errorMessages, []);
+  set(errorMessages, {});
   get(v$).$reset();
 };
 
 const clearErrors = () => {
-  const elements = get(errorMessages).length;
-  for (let i = 0; i < elements; i++) {
-    set(errorMessages, []);
-  }
+  set(errorMessages, {});
 };
 
 async function merge() {
   set(pending, true);
   const result = await mergeAssets({
-    sourceIdentifier: get(source),
-    targetIdentifier: get(target)
+    sourceIdentifier: get(sourceIdentifier),
+    targetIdentifier: get(targetIdentifier)
   });
 
   if (result.success) {
     set(done, true);
   } else {
-    set(errorMessages, [
-      ...get(errorMessages),
-      result.message ?? t('merge_dialog.error').toString()
-    ]);
+    set(
+      errorMessages,
+      typeof result.message === 'string'
+        ? { sourceIdentifier: result.message ?? t('merge_dialog.error') }
```

Comment:
```typescript
result.message ?? t('merge_dialog.error')
```
since at this point, `result.message` is a string, then `?? t('merge_dialog.error')` will never run

because it will only run when `result.message` is `null` or `undefined`

perhaps `|| t('merge_dialog.error')`

Reply:
true good catch, let me change that